content/docs/app-developer-guide/environment-variables.md

```diff
@@ -91,4 +85,41 @@ The following environment variables were set and available to buildpacks at buil
 
 > **NOTE:** Variables defined using `--env` take precedence over variables defined in `--env-file`.
 
+### Using Project Descriptor (`project.toml`, or `--descriptor`)
+The `--descriptor` parameter must be a path to a file which follows the project.toml [schema][descriptor-schema].
+You can define environment variables in an `env` table in the file, and pass those into the application.
+
+##### Example:
+```bash
+# clone the repo
+git clone https://github.com/buildpacks/samples
+
+# Add an environment variable to the project.toml
+printf "[[build.env]]\nname='HELLO'\nvalue='WORLD'" >> samples/apps/bash-script/project.toml
+
+# build the app
+pack build sample-app \
+    --builder cnbs/sample-builder:bionic \
+    --buildpack  samples/buildpacks/hello-world/ \
+    --buildpack samples/apps/bash-script/bash-script-buildpack/ \
+    --path  samples/apps/bash-script/
+
+# run the app
+docker run sample-app
+```
+
+The following environment variables were set and available to buildpacks at build-time:
+
+| Name    | Value   |  _Source_                  |
+|---------|---------|----------------------------|
+| `HELLO` | `WORLD` | _hard-coded value in file_ |
+
+<p class="spacer"></p>
+
+> **NOTE:** Variables defined using `--env` or `--env-file` take precedence over variables defined in the `project.toml`.
+
+> **NOTE:** `project.toml` can't detect environment variables (so, for instance, if one ran `export FOO=BAR` and added
+>`name=FOO` to the `project.toml`, it wouldn't detect any value set for `FOO`).
```

Comment:
I'm not sure if this is a feature of a bug. Should `project.toml` detect env vars set? Should I open a bug in pack?

Reply:
This would be a feature request. I don't believe this is an expectation and given that this is a [spec'd](https://github.com/buildpacks/spec/blob/main/extensions/project-descriptor.md#buildenv) it would require an RFC.